hardcoded.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+// Used to generate data for consts.php INSTALLER_TO_REPO_MINOR_VERSIONS
+// Copy paste in .cow.pat.json from the current release (file is .gitignored)
+
+include 'consts.php';
+
+$fn = '.cow.pat.json';
+if (!file_exists($fn)) {
+    echo "Create $fn\n";
+    die;
+}
+
+$versions = [];
+
+function repoName(string $name)
+{
+    $prefix = '';
+    if (!preg_match('#^(cwp/)#', $name) &&
+        !preg_match('#(/recipe-|/silverstripe-|/comment-notifications)#', $name)
+    ) {
+        $prefix = 'silverstripe-';
+    }
+    return $prefix . explode('/', $name)[1];
+}
+
+function parseNode(string $name, stdClass $node, array &$versions)
+{
+    $repoName = repoName($name);
+    if (!in_array($repoName, LOCKSTEPPED_REPOS) &&
+        !in_array($repoName, NO_INSTALLER_LOCKSTEPPED_REPOS) &&
+        !in_array($repoName, NO_INSTALLER_UNLOCKSTEPPED_REPOS)
+    ) {
+        preg_match('#^([0-9]+\.[0-9]+)#', $node->Version, $m);
+        $versions[$repoName] = preg_replace('#-(alpha|beta|rc)[1-9]+$#', '', $m[1]);
+    }
+    foreach ((array) $node->Items as $itemName => $item) {
+        parseNode($itemName, $item, $versions);
+    }
+}
+foreach (json_decode(file_get_contents($fn)) as $itemName => $item) {
+    parseNode($itemName, $item, $versions);
+}
+
+ksort($versions);
+foreach ($versions as $repoName => $version) {
+    echo "        '$repoName' => '$version',\n";
```

Comment:
Is this just echoing out the terminal? Why not have it update the consts file directly?

Reply:
We don't know what installer version to actually update.  I guess you could derive this from the version of installer being released, but really it's so trivial to just copy paste.  Also maybe someone already manually added some versions in.

Reply:
It _is_ trivial to copy paste, but not _as_ trivial as just going `git diff`, checking it looks right, and then `git commit -a` immediately.
Writing directly to the file (incl merging any existing values for the current version) can easily be done as an enhancement later if you're not keen on doing it right away though.

Reply:
Do later

Reply:
Created https://github.com/silverstripe/gha-generate-matrix/issues/21 for this purpose